[OSDOCS-11894] Explain why and in what context are the required GCP APIs used by OSD (#85242)
openshift/openshift-docs: modules/ccs-gcp-customer-procedure.adoc

. link:https://cloud.google.com/service-usage/docs/enable-disable#enabling[Enable] the following required APIs in the project that hosts your {product-title} cluster:
+
.Required API services
[cols="2a,3a,3a",options="header"]

|===

|API service |Console service name |Purpose

|link:https://cloud.google.com/deployment-manager/docs/apis#google-cloud-deployment-manager-v2-api[Cloud Deployment Manager V2 API]
|`deploymentmanager.googleapis.com`
|Used for automated deployment and management of infrastructure resources.

|link:https://cloud.google.com/compute/docs/reference/rest/v1[Compute Engine API]
|`compute.googleapis.com`
|Used for creating and managing virtual machines, firewalls, networks, persistent disk volumes, and load balancers.

// |link:https://cloud.google.com/apis/docs/overview[Google Cloud APIs]
// |`cloudapis.googleapis.com`
// |

|link:https://cloud.google.com/resource-manager/reference/rest[Cloud Resource Manager API]
|`cloudresourcemanager.googleapis.com`

> |Used for getting projects, getting/setting IAM policy for projects, validating required permission, and tagging.

Reviewer: ```suggestion
|Used for getting projects, getting or setting an IAM policy for projects, validating required permission, and tagging.
```

|link:https://cloud.google.com/iam/docs/reference/rest[Identity and Access Management (IAM) API]
|`iam.googleapis.com`
|Used for managing the IAM configuration for the cluster.

|link:https://cloud.google.com/service-infrastructure/docs/service-management/reference/rest[Service Management API]
|`servicemanagement.googleapis.com`
|Used indirectly to fetch quota information for GCP resources.

|link:https://cloud.google.com/service-usage/docs/reference/rest[Service Usage API]
|`serviceusage.googleapis.com`
|Used for determining what services are available in the customer’s Google Cloud account.

|link:https://cloud.google.com/storage/docs/json_api[Cloud Storage JSON API]
|`storage-api.googleapis.com`
|Used for accessing Cloud Storage for the image registry, ignition, and cluster backups (if applicable).

|link:https://cloud.google.com/storage/docs/apis[Cloud Storage]
|`storage-component.googleapis.com`
|Used for managing Cloud Storage for the image registry, ignition, and cluster backups (if applicable).

|link:https://cloud.google.com/resource-manager/docs/reference/orgpolicy/rest[Organization Policy API]
|`orgpolicy.googleapis.com`

> |Used to identify governance rules applied to customer’s Google Cloud that may impact cluster creation or management.

Reviewer: ```suggestion
|Used to identify governance rules applied to the customer’s Google Cloud that might impact cluster creation or management.
```

> |link:https://cloud.google.com/iap/docs/reference/rest[Cloud Identity-Aware Proxy API]
> |`iap.googleapis.com` ^[*]^
> |Used in emergency situations to troubleshoot cluster nodes that are otherwise inaccessible.

Reviewer 2: I see this is existing, but given that `*` could be a character in a URL and this `*` is an asterisk for a footnote, do you think moving it to the Purpose text makes sense? Or, since `[.small]` is technically not supported and we aren't supposed to use this anymore, maybe just bringing the statement into the box entirely?
```suggestion
|link:https://cloud.google.com/iap/docs/reference/rest[Cloud Identity-Aware Proxy API]
|`iap.googleapis.com`
|Used in emergency situations to troubleshoot cluster nodes that are otherwise inaccessible. This API is required for clusters deployed with Private Service Connect.
```

Author: Ah, good call, @jeana-redhat! I will move it in to the Purpose text. Thanks!!
